src/vivarium/framework/lifecycle.py

```diff
@@ -407,7 +407,7 @@ def to_guid(method: Callable[[Any], Any]) -> str:
         return f"{method.__self__.name}.{method.__name__}"  # type: ignore [attr-defined]
 
     def __call__(
-        self, method: Callable[[Any], Any], permitted_states: list[str]
```

Comment:
It's better to leave this signature as it was, and in `add_constraint()` convert the tuples to lists.